index.d.ts

```diff
@@ -68,7 +69,71 @@ declare namespace Eris {
   type MessageContent = string | AdvancedMessageContent;
   type MFALevel = 0 | 1;
   type PossiblyUncachedMessage = Message | { channel: TextableChannel | { id: string; guild?: Uncached }; guildID?: string; id: string };
-  type InteractionType = 1 | 2;
+
+  // Interaction
+  type InteractionType = 1 | 2 | 3;
+  type InteractionResponseType = 1 | 4 | 5 | 6 | 7;
+
+  type InteractionDataOptions = {
+    name: string;
+    type: SlashCommandOptionType;
+    value?: SlashCommandOptionType;
+    options?: InteractionDataOptions[];
+  };
+
+  type InteractionOptions = {
+    type: InteractionResponseType;
+    data?: InteractionContent;
+  };
+
+  type InteractionContent = {
+    allowedMentions?: AllowedMentions;
+    content?: string;
+    embeds?: EmbedOptions[];
+    flags?: number;
+    tts?: boolean;
+  };
+
+  type InteractionWebhookContent = Pick<WebhookPayload, "content" | "embeds" | "file" | "allowedMentions" | "tts" | "flags">;
+
+  //Slash Commands
+  type SlashCommandType = 1 | 2 | 3;
+
+  type SlashCommandOptionType = 1 | 2 | 3 | 4 | 5 | 6 | 7 | 8 | 9 | 10;
+
+  type SlashCommandOptions = {
+    type: SlashCommandOptionType;
+    name: string;
+    description: string;
+    required?: boolean;
+    choices?: { name: string; value: string | number};
```

Comment:
choices is an array
```suggestion
    choices?: { name: string; value: string | number }[];
```

Reply:
this should also have an options property, as top level options can have nested options (sub commands)